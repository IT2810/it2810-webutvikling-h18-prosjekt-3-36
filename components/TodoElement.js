import React from 'react';
import { Text, View, CheckBox } from 'react-native';

export default class TodoElement extends React.Component {
  render() {
    return (
    <View>
        <CheckBox
          title='Click Here'
          checked={true}
        />
      <Text>{this.props.text}</Text>
      <Text>{this.props.date}</Text>
      {this.props.completed && <Text>done</Text>}
    </View>
    );
  }
}

