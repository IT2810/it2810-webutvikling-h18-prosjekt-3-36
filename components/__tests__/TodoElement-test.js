import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import TodoElement from "../TodoElement";
import appState from '../../store/reducers/Reducer'
import { createStore } from 'redux';

const store = createStore(appState)

it("TodoElement Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TodoElement 
        key={1}
        text={"buy eggs"}
        date={"2018-10-07"}
        completed={false}
        category={"work"} 
        store={store} navigation={{ getParam: () => { }, setParams: () => { } }} isModalVisible={false} />);

    const renderer2 = new ShallowRenderer();
    renderer2.render(renderer.getRenderOutput());
    const result = renderer2.getRenderOutput();
    expect(result).toMatchSnapshot();
});