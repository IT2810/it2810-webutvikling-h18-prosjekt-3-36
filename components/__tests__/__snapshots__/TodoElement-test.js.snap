// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoElement Matches snapshot. 1`] = `
<Connect(TodoElement)
  category="work"
  completed={false}
  date="2018-10-07"
  text="buy eggs"
/>
`;
